### PR TITLE
feat: Truncate long collection descriptions and display a 'read more' link (#1938)

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers
**Functional:** 

@tihuan 

---

## Changes
- added to long collection descriptions a button to toggle between [Show More](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?node-id=2988%3A31078) and [Show Less](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?node-id=3584%3A31174) views.

### CODE REVIEW

File: frontend/src/components/Collection/components/CollectionDescription/index.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  const onUpdateMode = useCallback((): void => {
+    if (descriptionRef.current) {
+      const el = descriptionRef.current;
+      setEllipsisMode((currentMode) => getEllipsisMode(el, currentMode));
+    }
+  }, []);
+
+  useEffect(() => {
+    onUpdateMode(); /* Initialize ellipsis mode. */
+    window.addEventListener("resize", onUpdateMode);
```
Comment (tihuan) [inline frontend/src/components/Collection/components/CollectionDescription/index.tsx:54]: discussion: Is this a good use case for ResizeObserver 🙆‍♂️ ?

https://web.dev/resize-observer/

Sounds like there's perf upside!

Thank you!

**Review by tihuan — APPROVED** [review]
LGTM! Just one non blocking comment 😆 Thanks so much, Fran!! 🙏 ✨

File: frontend/src/components/Collection/components/CollectionDescription/index.tsx (hunk 1 above)
Comment (frano-m) [reply frontend/src/components/Collection/components/CollectionDescription/index.tsx:54]: oooooo, @tihuan, that's awesome...

**Review by tihuan — APPROVED** [review]
Woah this is even better!! Thank you so much for creating a hook for ResizeObserver 🤩 🙌 !! LGTM to the max 🚀 🏆 !!

**Comment by frano-m** [comment]
@tihuan thanks for the awesome review!
I've made some minor changes for clarity to the ResizeObserver hook, and added an early exit for `getEllipsisMode` in the `CollectionDescription` component - let me know if these final changes are ok!

**Pull request merged**